File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //Add letters if character is alphatecial
        if (isalpha(text[i]))
        {
            letters++;
        }

        //Add words if we found a whitespace
        else if (isspace(text[i]))
        {
            words++;
        }

        //Add sentences if we found . ! ?
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    //variables of the Coleman-Liau index
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    //The Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS50/pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("text: ")

letters = 0
words = 1
sentences = 0

for char in text:
    # Add letters if character is alphabetical
    if char.isalpha():
        letters += 1
    # Add words if we found a whitespace
    elif char.isspace():
        words += 1
    # Add senteces if we found . ! ?
    elif char in ['.', '!', '?']:
        sentences += 1

# Variables for the Coleman-Liau index
L = letters / words * 100
S = sentences / words * 100

# Coleman-Liau index formula
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
